### PR TITLE
[process-agent] Small Fixes for Process Discovery Check

### PR DESCRIPTION
**Pull request by just-chillin**
### What does this PR do?
This PR contains small fixes that didn't make it into https://github.com/DataDog/datadog-agent/pull/9082, namely:
- Fixed extra allocations for process collection on windows and linux
- Added message for when the user specifies a value below the minimum (currently 10m) for `process_config.process_discovery.interval`
 
### Motivation
The original PR (https://github.com/DataDog/datadog-agent/pull/9082) was holding back a couple of our teams other PR's, so we decided to go ahead with merging it and fix a couple of outstanding issues in a separate PR.

### Describe how to test your changes
- Set a value below `10m` for `process_config.process_discovery.interval` in your `datadog.yaml`, and make sure that a message appears saying it has been reset.
- Run the process and process_discovery check on linux and windows and ensure that it works as expected.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [x] The `need-change/operator` and `need-change/helm` labels has been applied if applicable.
- [x] The appropriate `team/..` label has been applied, if known.
- [x] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [x] The [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated if applicable.

Note: Adding GitHub labels is only possible for contributors with write access.

### CODE REVIEW

File: pkg/process/config/yaml_config.go [hunk 1]
```diff
@@ -303,8 +302,13 @@ func (a *AgentConfig) initProcessDiscoveryCheck() {
...
 	// We use a minimum of 10 minutes for this value.
-	a.CheckIntervals[DiscoveryCheckName] =
-		time.Duration(math.Max(float64(config.Datadog.GetDuration(key(root, "interval"))), float64(10*time.Minute)))
+	const discoveryMinInterval = 10 * time.Minute
+	if configuredInterval := config.Datadog.GetDuration(key(root, "interval")); configuredInterval >= discoveryMinInterval {
+		a.CheckIntervals[DiscoveryCheckName] = configuredInterval
+	} else {
+		a.CheckIntervals[DiscoveryCheckName] = discoveryMinInterval
+		log.Infof("Invalid interval for process discovery (<= %dm) using default value of %[1]d", discoveryMinInterval.Minutes())
+	}
```
Comment (xornivore) [inline pkg/process/config/yaml_config.go:311]: The code would read better if you avoid else here and more complex logic there. Effectively you are just resetting to the min interval. Disclaimer - I cannot guarantee the suggestion below compiles, apply at your own risk :) 
Other nits here:
- formatting of `time.Duration`, could you not just rely on builtin formatting?
- let's use `Warnf`
- We can define the minimal interval const at the top of the file along with other consts.
```suggestion
	const discoveryMinInterval = 10 * time.Minute
	discoveryInterval := config.Datadog.GetDuration(key(root, "interval"))
	if discoveryInterval < discoveryMinInterval {
		discoveryInterval = discoveryMinInterval
		log.Warnf("Invalid interval for process discovery (< %dm), using default value of %[1]d", discoveryMinInterval.Minutes())
	}
	a.CheckIntervals[DiscoveryCheckName] = discoveryInterval
```

Comment (just-chillin) [reply pkg/process/config/yaml_config.go]: Fixed in https://github.com/DataDog/datadog-agent/pull/9283/commits/4d7f0e5ecef70cab85a27ed31e2d8b498658072e

File: pkg/process/config/yaml_config.go [hunk 2]
```diff
@@ -303,8 +303,12 @@ func (a *AgentConfig) initProcessDiscoveryCheck() {
 
 	// We don't need to check if the key exists since we already bound it to a default in InitConfig.
 	// We use a minimum of 10 minutes for this value.
-	a.CheckIntervals[DiscoveryCheckName] =
-		time.Duration(math.Max(float64(config.Datadog.GetDuration(key(root, "interval"))), float64(10*time.Minute)))
+	discoveryInterval := config.Datadog.GetDuration(key(root, "interval"))
+	if discoveryInterval < discoveryMinInterval {
+		discoveryInterval = discoveryMinInterval
+		_ = log.Warnf("Invalid interval for process discovery (<= %sm) using default value of %[1]s", discoveryMinInterval.String())
```
Comment (xornivore) [inline pkg/process/config/yaml_config.go:309]: nit
```suggestion
		_ = log.Warnf("Invalid interval for process discovery (<= %s) using default value of %[1]s", discoveryMinInterval.String())
```

**Review by xornivore — COMMENTED** [review]
Small nit but lgtm otherwise :shipit:

File: pkg/process/config/yaml_config.go (hunk 2 above)
Comment (just-chillin) [reply pkg/process/config/yaml_config.go]: Oh! I didn't notice that I forgot to fix that. Thanks so much :)

**Review by xornivore — APPROVED** [review]

**Review by wiyu — APPROVED** [review]

**Review by vickenty — APPROVED** [review]

**Pull request merged**